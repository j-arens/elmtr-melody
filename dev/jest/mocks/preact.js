module.exports = {
    unmountComponent: jest.fn(),
};
