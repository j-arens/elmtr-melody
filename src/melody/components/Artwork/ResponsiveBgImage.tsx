import { WithOptionalClassName } from '@components/type';
import { TrackArtwork } from '@redux/type';
import { cySelector, getContextualTrackSizes } from '@utils/index';
import * as throttle from 'lodash.throttle';
import { Component, h } from 'preact';

interface Props extends WithOptionalClassName {
    artwork: TrackArtwork;
}

interface State {
    artworkSource: string;
}

export default class extends Component<Props, State> {
    state = {
        artworkSource: '',
    };

    componentDidMount() {
        this.setArtworkSource();
        window.addEventListener('resize', this.setArtworkSource);
    }

    componentWillUnmount() {
        window.removeEventListener('resize', this.setArtworkSource);
    }

    setArtworkSource = throttle(() => {
        const { artwork: { sizes, source_url } } = this.props;
        const relevantSizes = getContextualTrackSizes(sizes);

        if (relevantSizes.length) {
            this.setState({ artworkSource: relevantSizes[0].uri });
            return;
        }

        this.setState({ artworkSource: source_url });
    }, 1000);

    render({ className = '' }, { artworkSource }) {
        return (
            <figure
                class={className}
                style={{ backgroundImage: `url(${artworkSource})` }}
                {...cySelector('track-artwork')}
            />
        );
    }
}
