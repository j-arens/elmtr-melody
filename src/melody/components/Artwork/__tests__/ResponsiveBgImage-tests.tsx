import { h } from 'preact';
import { render } from 'preact-render-to-string';
import ResponsiveBgImage from '../ResponsiveBgImage';

describe('ResponsiveBgImage', () => {
    it('renders', () => {
        const component = render(
            <ResponsiveBgImage
                className="british-racing-green"
                artwork={{}}
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
