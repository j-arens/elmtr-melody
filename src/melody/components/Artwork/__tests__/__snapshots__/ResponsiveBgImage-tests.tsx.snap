// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveBgImage renders 1`] = `<figure class="british-racing-green" style="background-image: url();"></figure>`;
